libtock/defer.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define __DEFER__(F, V)                \
+        auto void F(int* V##SENTINEL); \
+        [[gnu::cleanup(F)]] int V;     \
+        auto void F(__attribute__ ((unused)) int* V##SENTINEL)
+
+#define defer __DEFER(__COUNTER__)
+#define __DEFER(N) __DEFER_(N)
+#define __DEFER_(N) __DEFER__(__DEFER_FUNCTION_##N, __DEFER_VARIABLE_##N)
```

Comment:
We should cite the source for this code. Probably also put a note that this mirrors the behavior of the C23 `defer` primitive.

(If I was looking at this without context I'd be like "what did those weird Tock people do???")

Reply:
I wrote something.

Reply:
Thanks!